app/services/questionnaire_package_import_service.rb

```
@@ -0,0 +1,556 @@
+  def read_uploaded_file(file)
+    return nil if file.blank?
+
+    file.respond_to?(:read) ? file.read : File.read(file.path)
+  ensure
+    file.rewind if file.respond_to?(:rewind)
+  end
+
+  # Reject unrelated or unsupported package versions before reading CSV rows.
+  def validate_package!(entries)
+    missing_files = REQUIRED_FILES - entries.keys
+    raise StandardError, "Questionnaire package is missing required files: #{missing_files.join(', ')}" if missing_files.any?
```

RuboCop (1.86.1)

[convention] Modifier form of `if` makes the line too long.

(Style/IfUnlessModifier)